### ISSUE TITLE
Calculate city/state from ZIP code rather than entering all three fields in patient information partial

### ISSUE THREAD

**Issue by colinxfleming** [DCAFEngineering/dcaf_case_management#306]
Right now we're asking for the following information from patients, including city, state, and zip:

![screen shot 2016-05-29 at 11 02 58 am](https://cloud.githubusercontent.com/assets/3866868/15634242/eb1f1870-258c-11e6-9ba3-1dc76c14c571.png)

We should be able to calculate city/state from just ZIP, so that CMs don't have to enter all three fields, and to reduce chances of weird typos mucking things up.

**Comment by colinxfleming** [comment]
Now that we've got some user behavior on record: Collecting zip seems to be spotty. We can talk with CMDs about whether it's appropriate to eschew city/state and autopopulate based on zip code. Marking this as a question for now until I figure that out with the CMDs.

The good news is: there appear to be ruby libraries we can pretty much plug in that are reasonably snappy!

**Comment by lwaldsc** [comment]
That would be really fantastic and save CMs time in addition to ensuring we
get that data. Would it work like "enter zip" then "pick from list of
towns"?

**Comment by colinxfleming** [comment]
@lwaldsc It'd probably be more like just 'enter zip' and then city/state would autopopulate. (There are a handful of zip codes with multiple cities or states, but even in those cases). I've got a privacy concern about storing ZIP though -- wonder if that's a little too specific than we absolutely need, esp since I don't think we'll ever really need to analyze on the zip level when city-state will do okay. (Though I guess that would make any hardcore data analysis difficult since most of the resources there are on the zip level.) What are your thoughts? 

Maybe we could build out a module that didn't persist ZIP and just saved city/state from ZIP instead, and just not save zip or let people enter city/state by hand so we can be sure to get valid data? 

That's kind of a lot of thinking out loud on github, but I think the tldr of it is: 
* I don't want to save zip unless we decide there isn't a privacy concern
* But I do want the advantages that entering a zip code provides, like normalized city/state
* So maybe the answer is to require zip code entry and to populate city/state from there, and just not save zip code

**Comment by colinxfleming** [comment]
I increasingly don't love the idea of storing zip code, which anecdotally it doesn't seem like our CMs are using that much, and which does trend toward the level. I also think that for clinic finder purposes we can either adjust it to take city/state or just have CMs be like 'we have this tool that helps people find closest clinics, what's your zip?' AND the libs we would crib off are not terribly efficient.

tldr I think zip code, and by extension this ticket, are a wontfix because we don't really need this.

@lwaldsc @NerdyGirl537 interested in your thoughts on this.

**Comment by NerdyGirl537** [comment]
Dude, I'm not even sure why we put it in to begin with.  It was never a field on the spreadsheet.  Maybe it was to make it more portable to other organizations?

**Comment by colinxfleming** [comment]
Going with 'nah' on this unless some fund or other needs to track zip. Closing.
